smartsim/_core/control/previewrenderer.py

```diff
@@ -100,6 +106,17 @@ def as_toggle(_eval_ctx: u.F, value: bool) -> str:
     return "On" if value else "Off"
 
 
+@pass_eval_context
+def get_ifname(_eval_ctx: u.F, value: t.List[str]) -> str:
+    return next((item for item in value if "ifname" in item), str).split("=")[-1]
```

Comment:
I think this will blow up if the filter clause fails. I think I either don't understand it, or the default `str` value could be a problem. Won't that pass the str _type_ and not an empty string?

I'd like to see tests for these methods added, w/edge case checks:
- `get_ifname`
- `get_dbtype `

Reply:
Try something _direct_, like:

```
tpl = jinja.template("foo | get_ifname")
rendered_preview = tpl.render(
    foo=None
)
```

Reply:
what edge cases should I check for?

Reply:
strings missing an equal, strings with just equal, strings with LHS=, strings w/=RHS, None